src/main/java/codeu/controller/FileUploadServlet.java

```diff
@@ -0,0 +1,41 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+public class FileUploadServlet extends HttpServlet {
+
+	@Override
+	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+		//get the file chosen by the user
+		Part filePart = request.getPart("fileToUpload");
+		
+		//get the InputStream to store the file somewhere
+	    InputStream fileInputStream = filePart.getInputStream();
+	    
+	    //for example, you can copy the uploaded file to the server
+	    //note that you probably don't want to do this in real life!
+	    //upload it to a file host like S3 or GCS instead
+	    File fileToSave = new File("/WEB-INF/view/profile.jsp" + filePart.getSubmittedFileName());
```

Comment:
It may make more sense to save the file in the mediaStore. I'll make the necessary changes to media so it is possible.